### DISCUSSION TITLE
Customized Table Output for Multiple Book Formats?

### DISCUSSION THREAD

**Discussion by SEthanMilne** [quarto-dev/quarto-cli#5236] [Q&A] [answered]
I was really excited to see Quarto 1.3 include support for outputting multiple book formats simultaneously!

My dissertation requires two outputs: a PDF document for official submissions, and a word document for advisor comments. 

I'd like to ensure consistency across output formats. However, I use {KableExtra} for table formatting for the PDF document, which as an HTML-type table does not translate into DOCX (even with YAML prefer-html: true). Is there a way to set custom options for table outputs such that, when outputting to multiple formats, different table outputs are used?

For example, I'd like the following to happen:

- A PDF document is outputted, which uses {KableExtra} tables for Table 1
- A DOCX document is outputted, which uses {gt} (or other!) tables for Table 1

As an alternate solution: is anyone aware of how to get KableExtra tables to properly output in DOCX format? A stopgap measuring could be saving the tables as a .png and loading them in as an image, but I'd ideally like the word doc version to be editable for advisor feedback.

Any help would be appreciated!

**Comment by cderv** [comment, marked as answer]
You get several things to tackle here: 

* How do you create your table ? 
* How do you target several output format ? 

One solution is to do both at the same time by using a Table 📦 that will know how to create table for different outputs. 
**kableExtra** knows that for HTML and PDF, but not for DOCX - from there doc, it does not support direct output to docx [without copy pasting from HTML to word](https://haozhu233.github.io/kableExtra/kableExtra_and_word.html)
So you need to use some 📦 that knows both format you want to target like [**flextable**](https://ardata-fr.github.io/flextable-book/) or [**gt**](https://gt.rstudio.com/). With those package you can write the same R code, including styling, and the package will know how to output to PDF and DOCX. 

If you really need to use **kableExtra** for HTML output, and want to use **gt** for other format, then you can conditional base on the output format using `knitr::pandoc_to()`

Either in one chunk using 
````r
if (knitr::pandoc_to("html")) {
  # do use kableExtra
} else {
  # do use gt
}
````

You could also do that with 2 chunks as shown in [R Markdown cookbook recipes for LaTeX or HTML](https://bookdown.org/yihui/rmarkdown-cookbook/latex-html.html)

[Conditional content is possible with Quarto ](https://quarto.org/docs/authoring/conditional.html)too, but this will just filter what is included in output, and all the code are still executed. Could work well for your usage though. 

Hope it helps

**Reply by SEthanMilne** [reply]
Thanks so much! This is very helpful. I do need KableExtra for some very odd formatting things, but I tested your conditional solution and it solved my problem :)
